File: JS/Prime_Palindrome.js
// Write a program which determines the largest prime palindrome less than 1000.
console.log(palin());

function palin() {
    for (var x = 999; x > 0; x--) {
        var pal = JSON.stringify(x).split('').reverse().join('');
        var not_prime = true;
        var is_palindrome = false;

        if (pal === x.toString()) {
            is_palindrome = true;
        }

        for (var i = 2, j = Math.floor(Math.sqrt(x)); i < j; i++) {
            if (x % i === 0) {
                not_prime = false;
            }
        }

        if (not_prime && is_palindrome) {
            return x;
        }
    }
}

File: Ruby/Prime_Palindrome.rb
# Write a program which determines the largest prime palindrome less than 1000.

def palin
  999.downto(0) do |x|
    pal = x.to_s.split('').reverse.join('')
    not_prime = true
    is_palindrome = false

    is_palindrome = true if pal == x.to_s

    j = Math.sqrt(x).floor

    (2..j).each do |i|
      not_prime = false if x % i == 0
    end
    return x if is_palindrome && not_prime
  end
end
